src/test/java/dk/sdu/se_f22/sortingmodule/range/rangepublic/TimeFilterTest.java

```diff
@@ -0,0 +1,387 @@
+            @DisplayName("Set invalid TimeFilter userMin higher than dbMin")
+            @MethodSource("provideTimeFilters")
+            void setInvalidTimeFilterUserMinHigherThanDBMin (TimeFilter filter) {
+                Instant newUserMin = Instant.ofEpochMilli(filter.getDbMaxInstant().toEpochMilli() + 10);
+                Assertions.assertThrows(IllegalMinMaxException.class,
+                        () -> filter.setUserMin(newUserMin)
+                );
+            }
+
+            @ParameterizedTest(name = "{0}")
+            @DisplayName("Set invalid TimeFilter userMax higher than dbMin")
+            @MethodSource("provideTimeFilters")
+            void setInvalidTimeFilterUserMaxHigherThanDBMin (TimeFilter filter) {
+                Instant newUserMax = Instant.ofEpochMilli(filter.getDbMaxInstant().toEpochMilli() + 10);
+                Assertions.assertThrows(IllegalMinMaxException.class,
+                        () -> filter.setUserMax(newUserMax)
+                );
+            }
+
+            @ParameterizedTest(name = "{0}")
+            @DisplayName("Set invalid TimeFilter userMax less than userMin")
+            @MethodSource("provideTimeFilters")
+            void setInvalidLessThanUserMaxLessThanUserMin (TimeFilter filter) throws IllegalMinMaxException {
+                Instant newUserMin = Instant.ofEpochMilli(filter.getDbMaxInstant().toEpochMilli()- 10);
+                Instant newUserMax = Instant.ofEpochMilli(filter.getDbMaxInstant().toEpochMilli()- 15);
+
+                filter.setUserMin(newUserMin);
+
+                Assertions.assertThrows(IllegalMinMaxException.class,
+                        () -> filter.setUserMax(newUserMax)
+                );
+            }
+
+
+            @ParameterizedTest(name = "{0}")
+            @DisplayName("Set invalid TimeFilter userMin higher than userMax")
+            @MethodSource("provideTimeFilters")
+            void setInvalidUserMinHigherThanUserMax (TimeFilter filter) throws IllegalMinMaxException {
+                Instant newUserMin = Instant.ofEpochMilli(filter.getDbMaxInstant().toEpochMilli()- 10);
+                Instant newUserMax = Instant.ofEpochMilli(filter.getDbMaxInstant().toEpochMilli()- 100);
+
+                filter.setUserMin(newUserMin);
+
+                Assertions.assertThrows(IllegalMinMaxException.class,
+                        () -> filter.setUserMax(newUserMax)
+                );
+            }
+
+            @ParameterizedTest(name = "{0}")
+            @DisplayName("Set userMin with incorrect type")
+            @MethodSource("provideTimeFilters")
+            void setUserMinWithIncorrectType (TimeFilter filter) {
+                double newValue = 100.0;
+                Assertions.assertThrows(IllegalMinMaxException.class,
+                        () -> filter.setUserMin(newValue)
+                );
+            }
+
+            @ParameterizedTest(name = "{0}")
+            @DisplayName("Set userMax with incorrect type")
```

Comment:
Naming concerns still applicable down to here

Reply:
done